Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```
@@ -51,113 +53,112 @@ if(SwiftCore_DIR)
 endif()
 
 include(FindPackageHandleStandardArgs)
+include(PlatformInfo)
+
+# This was loosely modelled after other find modules
+# (namely FindGLEW), where the equivalent parameter
+# is not stored in cache (possibly because we want
+# the project importing it to be able to
+# it "immediately")
+if(NOT DEFINED SwiftCore_USE_STATIC_LIBS)
+  set(SwiftCore_USE_STATIC_LIBS OFF)
+  if(NOT BUILD_SHARED_LIBS AND NOT APPLE)
+    set(SwiftCore_USE_STATIC_LIBS ON)
+  endif()
+endif()
 
 if(APPLE)
+  list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  list(APPEND SwiftCore_LIBRARY_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
   # When building for Apple platforms, SwiftCore always comes from within the
   # SDK as a tbd for a shared library in the shared cache.
-  find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-    HINTS
-      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
-  find_library(SwiftCore_IMPLIB
-    NAMES "libswiftCore.tbd"
-    HINTS
-      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
-  add_library(swiftCore SHARED IMPORTED GLOBAL)
-  set_target_properties(swiftCore PROPERTIES
-    IMPORTED_IMPLIB "${SwiftCore_IMPLIB}"
-    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
-  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
-    SwiftCore_IMPLIB SwiftCore_INCLUDE_DIR)
+  #list(APPEND SwiftCore_NAMES libswiftCore.tbd)
```

Left this commented while I was experimenting with generator expressions

```suggestion
  list(APPEND SwiftCore_NAMES libswiftCore.tbd)
```